### ISSUE TITLE
LosslessValue not working as excepted from Int to String value

### ISSUE THREAD

**Issue by mpintoQoQa** [marksands/BetterCodable#39]
Hi,

I've found an issue when using `@LosslessValue` from Int response.
We are currently upgrading our backend to a new API version and we need to handle both `Int` and `String` value for the `id` field.

To handle it properly we decided to set the `id` field with the `@LosslessValue` Property Wrapper.

### Issue

```
struct Response: Codable {
    @LosslessValue var id: String
}

let json = #"{ "id": 1 }"#.data(using: .utf8)!
let result = try JSONDecoder().decode(Response.self, from: json)

print(result) // Response(id: "true")
```
The result for this example should be `Response(id: "1")` instead of `Response(id: "true")`
Note that everything works as excepted for int value `> 1`

### Cause

This seems because of the decode priority in the `types` array in `LosslessValue.swift`

```
let types: [(Decoder) -> LosslessStringCodable?] = [
                decode(String.self),
                decodeBoolFromNSNumber(),
                decode(Bool.self),
                decode(Int.self),
                decode(Int8.self),
                decode(Int16.self),
                decode(Int64.self),
                decode(UInt.self),
                decode(UInt8.self),
                decode(UInt16.self),
                decode(UInt64.self),
                decode(Double.self),
                decode(Float.self)
            ]
```

As we can see the `decodeBoolFromNSNumber()` takes priority over the `decode(Int.self)`

**Comment by marksands** [comment]
Hmm, this isn't good at all. I may need to split up lossless values for boolean types. 🤔

**Comment by marksands** [comment]
@mpintoQoQa I am working on a [solution here](https://github.com/marksands/BetterCodable/compare/lossless-strategy?expand=1) if you want to take a look and see if this will work for your purposes. I plan on injecting LosslessValue with a strategy, similar to DefaultCodable, and will break out an additional strategy for the boolean priority for anyone that expects that behavior (with many caveats!).

/cc @serjooo since this will affect you (and who knows who else...) does that solution I linked work for you?

**Comment by serjooo** [comment]
Checked out the changes and everything looks fine from my side. Thank you for the change Mark!

**Comment by mpintoQoQa** [comment]
@marksands I've tested your solution in my project and everything works as expected. Your [solution](https://github.com/marksands/BetterCodable/compare/lossless-strategy?expand=1) looks great.
Thank you for the change.

**Comment by marksands** [comment]
Thanks to the both of you. I'll work on cleaning this up and adding docs as soon as I can to get this released.

**Comment by marksands** [comment]
Just released 0.4.0 https://github.com/marksands/BetterCodable/releases/tag/0.4.0 let me know if you experience any issue with it!
